samples/nextjs/scripts/generate-config.ts

```diff
@@ -33,17 +33,30 @@ export function generateConfig(configOverrides?: { [key: string]: string }): voi
 
   // The GraphQL endpoint is an example of making a _computed_ config setting
   // based on other config settings.
-  addGraphQLConfig(config);
-
-  const configText = `/* eslint-disable */
+  const computedConfig: { [key: string]: string } = {};
+  computedConfig.graphQLEndpoint =
+    '`${config.sitecoreApiHost}${config.graphQLEndpointPath}?sc_apikey=${config.sitecoreApiKey}`';
+
+  // Create an empty object defining all our configuration properties, so module export has proper typing
+  const configProps = Object.assign({}, config, computedConfig);
+  Object.keys(configProps).forEach((prop) => (configProps[prop] = ''));
```

Comment:
Since you got rid of the `BaseConfig` type and usage, is there still a reason we need this "blank" object? I think this generated output would work:
```
/* eslint-disable */
// Do not edit this file, it is auto-generated at build time!
// See scripts/bootstrap.ts to modify the generation of this file.
var config = {};
config.sitecoreApiKey = process.env.sitecoreApiKey || "2a434cac-2bcd-4534-a4f9-6593eea61f6e",
config.sitecoreApiHost = process.env.sitecoreApiHost || "https://cm.nextjs.localhost",
config.jssAppName = process.env.jssAppName || "rendering",
config.graphQLEndpointPath = process.env.graphQLEndpointPath || "/api/rendering",
config.graphQLEndpoint = `${config.sitecoreApiHost}${config.graphQLEndpointPath}?sc_apikey=${config.sitecoreApiKey}`;
module.exports = config;
```
```suggestion
```

Reply:
Fixed